cumulus_library/template_sql/codeable_concept_denormalize.sql.jinja

```diff
@@ -46,21 +62,28 @@ CREATE TABLE {{ target_table }} AS (
             {%- if filter_priority %}
             '{{ loop.index0 }}' AS priority,
             {%- endif %}
-            u.codeable_concept.code,
-            u.codeable_concept.display,
-            u.codeable_concept.system AS code_system
+            {%- for extra_field in extra_fields %}
+            {%- if extra_alias_done %}
+            {{ extra_field[1] }},
+            {%- else %}
+            s.{{ extra_field[0] }} AS {{ extra_field[1] }},
+            {%- endif %}
+            {%- endfor %}
+            u.coding.code,
+            {#- Prefer coding.display, but fall back to concept.text #}
+            COALESCE(u.coding.display, s.{{ field_alias }}.text) AS display,
+            u.coding.system AS code_system,
```

Comment:
**Goals of display/text**

Well, let's get alignment on the point of these labels. In my head, it's mostly just so we have something to show in the dashboard as presentation.

On a clinical level, I think we want studies to look at codes & systems (where possible). Trying to parse `display` or `text` will always be fraught across EHRs etc.

There is a big loophole in that "where possible", granted. But those values will be always be so fuzzy, I'm having a hard time imagining a case where the study knows the good stuff is in `text` but not `display`.

**Scenarios**
What's your vibes on each of these?

1. Display but no text: easy, we want to use display
2. Display and text: in my head, display will always be more relevant to the code in question
3. No display but text: I think using `text` in the CUBE is more friendly than `cumulus__none` and probably what the EHR intended consumers to do (you say this is unlikely, and I'm sure we could get stats from Cerner at least, but I don't put anything past EHRs)
4. No codings at all, but text: the "Cerner contained Medication" scenario and honestly, plenty of other places - I figured we'd want these to show up in the CUBE and be available for studies to attempt to parse if they want.

Reply:
Even if we want to eventually expose `text` alongside `display`, that is arguably a separate question from using `text` as a fallback for `display` today - like, in 3 and 4 above - I'd rather have a fallback than nothing.

Reply:
Another question: will 4 flood CUBEs with lots of one-off random texts? I dunno how EHRs populate those kind of uncoded fields. Like, if it's usually a dropdown (no problem then) or user-entered text (which seems like it could be a problem).

So in that case, maybe we don't want the fallback?

Reply:
1. easy, agree
2. agree
3. I agree with this outside of core.

inside of core, I think it's more nuanced. We are still trying to do a bit of 'here's the representation of your data as we found it' presentation here - we've backed out coalescing to `'None'` in favor of `'cumulus_none'` for this reason. This feels like going back to that same well of massaging data in core rather than extracting & re-organizing it. But I think we let a study author basically decide which approach makes  sense to them.

4 I think you're right in that 4 will probably generate a wall of junk in cubes, and more likely cause them to get dropped on the floor more than anything else. But I would leave them in for the same detail reasons in 3.

Re: just getting these from a know good source - we might be close with the file upload stuff i'm working on, so maybe we can make this problem go away before too long.

Reply:
> We are still trying to do a bit of 'here's the representation of your data as we found it' presentation here

There is a bit of tension here between this usage of `core` and the fact that `core` itself also generates counts tables - like for its own counts table, it probably needs an opinion on this kind of coalescing that it doesn't need to be a source of truth for studies.

So I'm hearing "let's not do this coalescing for now" from you - as well as a "maybe we'll add text in the future? who knows"

I can easily take out the scenario 3 coalesce() - that was a whole cloth invention of this PR.

But scenario 4 was used in for contained Cerner Medications. (I think Cerner also does that in other places, like Device.type etc) Do we want to leave it in place for that one Medication use case, or take a stance that we don't do that, and if you want that data, wait for us to do the `text` thing?

Reply:
And by "Do we want to leave it in place for that one Medication use case" - I mean, only do it for that one code path, and not do it for general codeable concepts.

Reply:
> There is a bit of tension here between this usage of core and the fact that core itself also generates counts tables - like for its own counts table, it probably needs an opinion on this kind of coalescing that it doesn't need to be a source of truth for studies.

I have been thinking of core counts like this:
- If you're generating core counts, it's the first time you're looking at the data through whatever version of the library you're running, for a given set of ETL extracts
- There are two uses for core tables - one is for cross-site population statistics, the other is as a 'what does this data even look like' check
  - The latter makes preserving nulls from the source system as is valuable in counts; coalescing them dilutes that value
  
So yeah, I'm in favor of removing number 3.

The cerner contained thing I don't think we can make a call on changing without at least Jamie in the loop - since he's out this week, I'd propose leaving it as is and making a spike ticket so we remember to discuss it?

Reply:
Talked a bit more in slack on this. I'll remove both 3 and 4 from this PR and this means this PR removes a tiny feature from the `core__medication` table, which is that contained Medications aren't represented anymore.

They were only used by Cerner, were not in any CUBEs exported by the `core` study, and weren't really usable (just text, no code).

Until we figure out how to approach `text`, this felt like the sanest/most-consistent path.

Further text thinking issue filed here: https://github.com/smart-on-fhir/cumulus-library/issues/229